javalib/src/main/scala/java/net/Socket.scala

```diff
@@ -248,7 +248,7 @@ class Socket protected (private[net] val impl: SocketImpl,
     val value = if (on) {
       if (linger > 65535) 65535 else linger
     } else {
-      -1
+      0
```

Comment:
@alexdupre Wonderful to see this contribution.
I am an individual contributor, not a Scala Center person.
My background is in network programming (implementing
TCP, UDP, etc in a commercial O/S kernel.)

When I saw this changes, I immediately tweaked.  After a while
studying the change, the Open Group Specification, Java JVM practice 
(not source), and the underlying SN PlainSocketImpl.scala, I believe
this change will not have a negative impact on existing code. But then
you know that to begin with.  

As you are probably well aware. socket code is particularly fragile,
especially since it is hard to test if any given change is innocuous
and end users get cranky when their years old application stops working
after an upgrade.  Forgive me if I am preaching to the choir (saying something that is well known and already believed.)

I will review the rest of this PR as soon as I can.  If I do not get there
before Scala Center review, I do not want to hold this PR up.

Reply:
For reference, this is the code the breaks (and throws a SocketException) when the value is negative: https://github.com/freebsd/freebsd-src/blob/098dbd7ff7f3da9dda03802cdb2d8755f816eada/sys/kern/uipc_socket.c#L3017-L3022

And this is the Java native implementation that sets 0 when SO_LINGER is disabled: https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/src/solaris/native/java/net/PlainSocketImpl.c#L944-L951

But thanks to your comment I found out that my initial patch wasn't fully correct, and I also found another issue in the code. I've just pushed the final fix.